src/Foundatio.Repositories.Elasticsearch/Configuration/DailyIndex.cs

```diff
@@ -45,11 +45,11 @@ public DailyIndex(IElasticConfiguration configuration, string name, int version
                 switch (document) {
                     case null:
                         throw new ArgumentNullException(nameof(document));
-                    case IHaveCreatedDate createdDoc when createdDoc.CreatedUtc != DateTime.MinValue:
-                        return createdDoc.CreatedUtc;
                     // This is also called when trying to create the document id.
                     case IIdentity identityDoc when identityDoc.Id != null && ObjectId.TryParse(identityDoc.Id, out var objectId) && objectId.CreationTime != DateTime.MinValue:
                         return objectId.CreationTime;
+                    case IHaveCreatedDate createdDoc when createdDoc.CreatedUtc != DateTime.MinValue:
```

Comment:
I moved these around because it aligns with the logic we are doing in other spots (where we have this exact expression)

Reply:
I guess this is alright. If it has an id it should probably prefer that.